### ISSUE TITLE
Aurora Module -- Creating Instance From Snapshot Fails on 1.30

### ISSUE THREAD

**Issue by jseiser** [terraform-providers/terraform-provider-aws#5450]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.11.7

### Affected Resource(s)

* aws_rds_cluster
* aws_rds_cluster_instance

### Terraform Configuration Files

```hcl
resource "aws_rds_cluster" "cluster" {
  cluster_identifier_prefix    = "${format("%s-%s-%s", var.customer, var.project, var.environment)}"
  engine                       = "${var.engine}"
  engine_version               = "${var.engine_version}"
  database_name                = "${var.database_name}"
  master_username              = "${var.username}"
  master_password              = "${var.password}"
  backup_retention_period      = "${var.backup_retention_period}"
  preferred_backup_window      = "${var.preferred_backup_window}"
  preferred_maintenance_window = "${var.preferred_maintenance_window}"
  vpc_security_group_ids       = ["${aws_security_group.aurora.id}"]
  apply_immediately            = "${var.apply_immediately}"
  db_subnet_group_name         = "${aws_db_subnet_group.db_subnet_group.id}"
  final_snapshot_identifier    = "${format("rds-%s-aurora-%s-%s", var.customer, var.project, var.environment)}"
  skip_final_snapshot          = true
  storage_encrypted            = "${var.storage_encrypted}"
  snapshot_identifier          = "${var.snapshot_identifier}"

  lifecycle {
    create_before_destroy = true
  }

  tags {
    Name        = "${format("rds-%s-aurora-%s-%s-%s", var.customer, var.project, var.function, var.environment)}"
    Customer    = "${var.customer}"
    Environment = "${var.environment}"
    Terraform   = true
    Database    = "Aurora"
  }
}

resource "aws_rds_cluster_instance" "cluster_instances" {
  count                      = "${var.instance_count}"
  identifier                 = "${format("rds-%s-aurora-%s-%s-%d", var.customer, var.project, var.environment, count.index)}"
  engine                     = "${var.engine}"
  engine_version             = "${var.engine_version}"
  cluster_identifier         = "${aws_rds_cluster.cluster.id}"
  instance_class             = "${var.instance_class}"
  apply_immediately          = "${var.apply_immediately}"
  auto_minor_version_upgrade = "${var.auto_minor_version_upgrade}"

  lifecycle {
    create_before_destroy = true
  }

  tags {
    Name        = "${format("rds-%s-aurora-%s-%s-%s-%d", var.customer, var.project, var.function, var.environment, count.index)}"
    Customer    = "${var.customer}"
    Environment = "${var.environment}"
    Project     = "${var.project}"
    Owner       = "${var.owner}"
    Function    = "${var.function}"
    Terraform   = true
  }
}
```

### Expected Behavior

Cluster and instance should have been created from the snapshot.

### Actual Behavior

Cluster created, but its not managed.  Another plan attempts to create another one.

```
Error: Error applying plan:

1 error(s) occurred:

* module.aurora.aws_rds_cluster.cluster: 1 error(s) occurred:

* aws_rds_cluster.cluster: InvalidParameterValue: The parameter ResourceName must be provided and must not be blank.
	status code: 400, request id: fa52736f-c76c-450a-a1fb-7f23042b59fd

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### Steps to Reproduce

Upgrade aws-provider to 1.30.0
terraform apply

### References

* https://github.com/terraform-providers/terraform-provider-aws/issues/5360

**Comment by myoung34** [comment]
This also happens in non clusters, just an `aws_db_instance`. Downgraded version fixed it

**Comment by bflad** [comment]
@myoung34 can you please open a separate bug report with full details of your `aws_db_instance` configuration? I had written [an acceptance test](https://github.com/terraform-providers/terraform-provider-aws/pull/5370/files#diff-0f423e325a63ba691f526a09026a9bbdR422) to cover this error within `aws_db_instance` and verify the fix, but maybe there are other parts to it.

**Comment by myoung34** [comment]
@bflad done in #5477

**Comment by bflad** [comment]
Fix for `aws_rds_cluster` with `snapshot_identifier` and `tags` submitted: #5479

**Comment by bflad** [comment]
The fix for `aws_rds_cluster` has been merged and will release with version 1.31.0 of the AWS provider, likely later today. 👍
